### PR TITLE
Adding averaging of iterates

### PR DESCRIPTION
**Pull request by Tomerporian**
Supports EMA and polynomial averaging. Includes option for more than one averaged model and taking intermittent averaging.
For example, running EMA with parameter 0.999 and period 100 and polynomial averaging with parameter 8 and period 10, use
`--averagers="ema_0.999_100,poly_8_10"`
Logging of training loss of the averaged models can be done by choosing `--log-avg-model-training-loss` to be bigger than zero.

### CODE REVIEW

File: open_lm/main.py [hunk 1]
```diff
@@ -636,14 +668,27 @@ def main(args):
...
+            scheduler = const_lr(
+                optimizer,
+                args.lr,
+                args.warmup,
+                # total_steps,
+                # args.lr_cooldown_end,
+                # args.force_min_lr,
+            )
+        else:
+            logging.error(f"Unknown scheduler, {args.lr_scheduler}. Available options are: cosine, const.")
```
Comment (achalddave) [inline open_lm/main.py:690]: Can you make this raise a ValueError exception instead?

File: open_lm/params.py [hunk 2]
```diff
@@ -237,7 +237,7 @@ def check_args(args):
             if args.remote_sync_protocol != "s3":
                 raise ValueError("Sync protocol not supported when using resume latest.")
 
-    if args.lr_scheduler != "cosine":
+    if args.lr_scheduler != "cosine" or args.lr_scheduler != "const" or args.lr_scheduler != "const-cooldown":
```
Comment (achalddave) [inline open_lm/params.py:240]: nit: Can you replace with the below for readability?
```if args.lr_scheduler not in {"cosine", "const", "const-cooldown"}:```

**Review by achalddave — COMMENTED** [review]
Overall this looks great, thanks @Tomerporian! Left a few small comments. I'll run a 1B model with a few settings for averagers, and assuming that goes smoothly, we should be good to merge!

File: open_lm/params.py (hunk 2 above)
Comment (achalddave) [reply open_lm/params.py:240]: actually the current version also has a bug which causes the unit tests to fail. more generally, maybe we can move the lr_scheduler components out of this PR?

File: open_lm/main.py (hunk 1 above)
Comment (Tomerporian) [reply open_lm/main.py]: sure no problem

File: open_lm/params.py (hunk 2 above)
Comment (Tomerporian) [reply open_lm/params.py]: i added the line as you wrote

**Review by achalddave — APPROVED** [review]
